rir/src/interpreter/interp.cpp

```diff
@@ -23,34 +23,37 @@ extern Rboolean R_Visible;
 using namespace rir;
 
 struct CallContext {
-    CallContext(Code* c, unsigned callsiteId, SEXP callee,
-                R_bcstack_t* eagerArgs, SEXP callerEnv, Context* ctx)
-        : eagerArgs(eagerArgs), caller(c->function()), callerEnv(callerEnv),
+    CallContext(Code* c, unsigned callsiteId, SEXP callee, bool hasEagerArgs,
```

Comment:
I find this a bit confusing. In RIR we have instructions that can be `*_lazy_*` and also `*_stack_*`, whereas in PIR we have `*Eager*`.
This context class uses eager and stack args, and eager callee.

Is the following accurate?
- `call_` is not supposed to be generated by PIR
- all args are passed on the stack after PIR
- `Call` ~ `call_stack_lazy_`
- `EagerCall` ~ `call_stack_`
- `StaticCall` ~ `static_call_stack_lazy_`
- `StaticEagerCall` ~ `static_call_stack_`
- in `CallContext`, `hasEagerArgs == true` implies `stackArgs != nullptr`

Summary: in RIR, the `lazy` instructions mean, that the args on the stack are promised, in PIR the `Eager` means the args are values, and `hasEagerCallee` says whether the callee expects values or promises?

Sorry, just trying to make sense of it.

Reply:
That is all correct, except, `call_` can be generated by pir. My idea was that we have pass that is scheduled late, that converts `Call`, where none of the values are eager, into a (new) `LazyCall` which translates to `call_`.

I agree the naming is a bit confusing. I would like to keep `Call` in pir, because the idea is, that `Call` can be anything, we don't know yet. If all are eager we convert to eager, if all are lazy, we convert to lazy. But probably `call_stack_lazy` is not a good name, how about `call_stack_promised`?

I think we cannot fully harmonize the names, because fundamentally pir and rir have a different style of describing a call. How about this:

* `Call` -> `call_stack_promised`   (mixed lazy and eager, but all promised)
* `EagerCall` -> `call_stack_eager`
* `LazyCall` -> `call_promised`

Reply:
(though renaming `call_` might be a tedious task :) )

Reply:
I am leaving `call_` as is for now

Reply:
Sure, thanks for shedding some light on this. The `promised` naming looks better to me